[REVIEW] Fix  MNMG failures in mg_dgl_extensions

VibhuJawa opened the pull request:
This PR fixes the below errors that have popped up in MNMG testing. 
- [x] fix_out of index keys on MNMG graphs
- [x] fix loc/get_node_storage error  on MNMG graphs 
(Work around  https://github.com/rapidsai/cudf/issues/11877)
- [x] Clear Cached Properties when they become invalid
- [x] Remove  6 pytest skipping as both these PRs have landed 
- https://github.com/rapidsai/cugraph/pull/2751
- https://github.com/rapidsai/cugraph/pull/2523
- [x] Change `vertex_col_names` to  `node_col_names`  to match DGL 
- [x] Ensure MNMG tests pass 
- [x] Work around the PG bug and also prevent redundant conversion to lists

VibhuJawa commented on python/cugraph/cugraph/dask/structure/mg_property_graph.py:
```diff
@@ -485,11 +484,6 @@ def get_vertex_data(self, vertex_ids=None, types=None, columns=None):
         if self.__vertex_prop_dataframe is not None:
             df = self.__vertex_prop_dataframe
             if vertex_ids is not None:
-                if isinstance(vertex_ids, int):
-                    vertex_ids = [vertex_ids]
-                elif not isinstance(vertex_ids,
-                                    (list, slice, self.__series_type)):
-                    vertex_ids = list(vertex_ids)
```
I think we should let the  underlying library handle it rather than limiting `location` to these types. 

Also, this fails with GPU objects like `cupy` , `cudf.Series` , for numpy arrays we pay a cost.

VibhuJawa commented on python/cugraph/cugraph/dask/structure/mg_property_graph.py:
```diff
@@ -721,11 +715,6 @@ def get_edge_data(self, edge_ids=None, types=None, columns=None):
-                    edge_ids = [edge_ids]
-                elif not isinstance(edge_ids,
-                                    (list, slice, self.__series_type)):
-                    edge_ids = list(edge_ids)
```
I think we should let the  underlying library handle it rather than limiting `location` to these types. 

Also, this fails with GPU objects like `cupy` , `cudf.Series` , for numpy arrays we pay a cost.

VibhuJawa replied:
Fails when fetching a single element.

VibhuJawa commented on python/cugraph/cugraph/structure/property_graph.py:
```diff
@@ -497,11 +497,6 @@ def get_vertex_data(self, vertex_ids=None, types=None, columns=None):
         if self.__vertex_prop_dataframe is not None:
             df = self.__vertex_prop_dataframe
             if vertex_ids is not None:
-                if isinstance(vertex_ids, int):
-                    vertex_ids = [vertex_ids]
-                elif not isinstance(vertex_ids,
-                                    (list, slice, self.__series_type)):
-                    vertex_ids = list(vertex_ids)
```
I think we should let the  underlying library handle it rather than limiting `location` to these types. 

Also, this fails with GPU objects like `cupy` , `cudf.Series` , for numpy arrays we pay a cost.

VibhuJawa replied:
Fails when fetching a single element.

VibhuJawa commented on python/cugraph/cugraph/structure/property_graph.py:
```diff
@@ -746,11 +741,6 @@ def get_edge_data(self, edge_ids=None, types=None, columns=None):
         if self.__edge_prop_dataframe is not None:
             df = self.__edge_prop_dataframe
             if edge_ids is not None:
-                if isinstance(edge_ids, int):
-                    edge_ids = [edge_ids]
-                elif not isinstance(edge_ids,
-                                    (list, slice, self.__series_type)):
-                    edge_ids = list(edge_ids)
```
I think we should let the  underlying library handle it rather than limiting `location` to these types. 

Also, this fails with GPU objects like `cupy` , `cudf.Series` , for numpy arrays we pay a cost.

rlratzel commented on python/cugraph/cugraph/gnn/graph_store.py:
```diff
@@ -462,7 +604,10 @@ def fetch(self, indices, device=None, pin_memory=False, **kwargs):
             subset_df = subset_df.compute()
-        tensor = self.from_dlpack(subset_df.to_dlpack())
+        if len(subset_df) == 0:
+            raise ValueError("indices={indices} not found in FeatureStorage")
```
```suggestion
            raise ValueError(f"{indices=} not found in FeatureStorage")
```

rlratzel reviewed (changes requested):
Thanks for adding the extra test coverage.  Just had a few minor suggestions/fixes/typos

VibhuJawa commented on python/cugraph/cugraph/gnn/graph_store.py:
```diff
@@ -51,24 +52,129 @@ def __init__(self, graph, backend_lib="torch"):
+            (ignored if is_single_vector_feature=False)
+        etype : string
+            The edge type to be added. This should follow the string format
+            '(src_type),(edge_type),(dst_type)'
```
So i enforce this in the DGL code upstream that calls the wrapper. 

https://github.com/rapidsai/dgl/blob/3dbb4c02d41a588bdb85d4445d1e891694199ab6/python/dgl/contrib/cugraph/cugraph_storage.py#L583-L590

VibhuJawa commented on python/cugraph/cugraph/gnn/graph_store.py:
```diff
@@ -51,24 +52,129 @@ def __init__(self, graph, backend_lib="torch"):
+            interface.
+        node_col_name : string
+            The column name that contains the values to be used as vertex IDs.
+        feat_name : string
```
Yup, I mention it in the doctoring below. 
https://github.com/rapidsai/cugraph/blob/e622fb2236725d73812b2010050724a880d6db32/python/cugraph/cugraph/gnn/graph_store.py#L75

VibhuJawa replied:
Actually expanded it more: 

https://github.com/rapidsai/cugraph/blob/d74cf87106d965cd030b303e5586aebdc8b8dc45/python/cugraph/cugraph/gnn/graph_store.py#L73-L76

VibhuJawa replied:
Thanks for the suggestion.

VibhuJawa commented:
> Thanks for adding the extra test coverage. Just had a few minor suggestions/fixes/typos

Thanks for the review Rick, Addressed all of your comments so should be ready for another round of reviews.

rlratzel commented:
@gpucibot merge
